hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -250,10 +252,12 @@ private void sendBatch(Map<Integer, List<Data>> batch, boolean replaceExistingVa
             futures.add(opService.invokeOnPartition(SERVICE_NAME, op, partitionId));
         }
 
-        LoadAllOperation op = new LoadAllOperation(mapName, Collections.<Data>emptyList(), replaceExistingValues, lastBatch);
-
         // notify SENDER_BACKUP
-        futures.add(opService.createInvocationBuilder(SERVICE_NAME, op, mapNamePartition).setReplicaIndex(1).invoke());
+        InternalPartition partition = partitionService.getPartition(mapNamePartition);
+        if (hasBackup && partition.getReplicaAddress(1) != null) {
```

Comment:
Found a better way. Now instead of checking replica address, checking for cluster size. In the original issue there was a single node and sending to replicas was generating an exception.